Implementation of component-test blueprint

alexander-alvarez opened the pull request:
<details>
<summary> Old description </summary>
I pretty much gutted https://github.com/emberjs/ember.js/tree/master/blueprints/component-test and adjusted to the best of my knowledge. 
I didn't know what to do for unit tests.

What we get from running `ember g component-test my-component -u` and then the codemod is:
```
import { module, test } from 'qunit';
import { setupTest } from 'ember-qunit';

module('Unit | Component | my component', function(hooks) {
  setupTest(hooks);

  test('it renders', function(assert) {
    
    // Creates the component instance
    /*let component =*/ this.owner.factoryFor('component:my-component').create();
    // Renders the component to the page
    this.render();
    assert.equal(this.$().text().trim(), '');
  });
});
```

- Do we want this? In that case, I can adjust the current implementation.

- As a general rule do we want the equivalent running the codemod on our current blueprints? or is now a time to change the format?

- Also what's the best way to test the blueprints?

**/cc @rwjblue @bendemboski @tbieniek**
</details> 


Related to https://github.com/emberjs/ember.js/issues/15933, I stole implementations and testcases from  https://github.com/emberjs/ember.js/pull/15934

Commit:
Implementation of component-test blueprint

rwjblue commented on package.json:
```diff
@@ -26,12 +26,16 @@
     "ember-cli": "~2.16.2",
     "ember-cli-dependency-checker": "^2.0.0",
     "ember-cli-eslint": "^4.0.0",
+    "ember-cli-get-component-path-option": "^1.0.0",
```
These deps added need to be in `dependencies` section (not devDeps)

rwjblue reviewed (commented):
I'll need to play around a bit with it to confirm the exact output, but this looks good!

Commit:
Move dev deps to deps

Turbo87 commented:
@alexander-alvarez it would probably be good to use https://github.com/ember-cli/ember-cli-blueprint-test-helpers to test the output of the new blueprints

alexander-alvarez commented:
note to self -- update to match implementation from https://github.com/emberjs/ember.js/pull/15934

Commit:
Add ember-cli-blueprint-test-helpers

Commit:
Setup test-runner & component-test blueprint

Shamelessly stolen from implementation in https://github.com/emberjs/ember.js/pull/15934

alexander-alvarez commented:
@Turbo87 updated with your changes that are in `ember.js#master` and test cases

rwjblue reviewed (commented):
Seems good to me...

Commit:
Update clone

Commit:
Fixup whitespace
